---
title: Errore del compilatore CS0156
ms.date: 07/20/2015
f1_keywords:
- CS0156
helpviewer_keywords:
- CS0156
ms.assetid: 32026b1b-bcd7-4464-b63f-3b38c00452a6
ms.openlocfilehash: 7ea5e861b71f5f65231bb29537a28a23a0821cef
ms.sourcegitcommit: 10986410e59ff29f2ec55c6759bde3eb4d1a00cb
ms.translationtype: MT
ms.contentlocale: it-IT
ms.lasthandoff: 05/31/2019
ms.locfileid: "66422970"
---
# <a name="compiler-error-cs0156"></a>Errore del compilatore CS0156
L'utilizzo dell'istruzione throw senza argomenti non è consentito in una clausola finally annidata all'interno della clausola catch di inclusione più vicina  
  
 Un'istruzione [throw](../../csharp/language-reference/keywords/throw.md) senza parametri può essere usata solo in una clausola **catch** che non accetta parametri.  
  
 Per altre informazioni, vedere [eccezioni e gestione delle eccezioni](../../csharp/programming-guide/exceptions/index.md).  
  
 L'esempio seguente genera l'errore CS0156:  
  
```csharp  
// CS0156.cs  
using System;  
  
namespace MyNamespace  
{  
   public class MyClass2 : Exception  
   {  
   }  
  
   public class MyClass  
   {  
      public static void Main()  
      {  
         try  
         {  
            throw;   // CS0156  
         }  
  
         catch(MyClass2)  
         {  
            throw;   // this throw is valid  
         }  
      }  
   }  
}  
```
